src/cavendish_particle_tracks/_magnification_dialog.py

```diff
@@ -16,12 +20,16 @@
 )
 from ._calculate import magnification
 
+if TYPE_CHECKING:
+    from ._main_widget import ParticleTracksWidget
+
 
 class MagnificationDialog(QDialog):
     def __init__(self, parent=None):
-        super().__init__(parent)
-
-        self.parent = parent
+        super().__init__(parent)  # Call QDialog constructor
+        self.parent: ParticleTracksWidget = (
+            parent  # Assign Napari viewer as parent.
+        )
```

Comment:
Can we get around this by putting the type hints in the argument?

```python
def __init__(self, parent: QWidget | None = None):
    super().__init__(parent)
    self.parent = parent
```

Actually, now I come to think of it, does the super classes constructor not set the parent for us? TBC

Reply:
Using :QWidget as a type hint would stop the type hinter from recognising particle-tracks-widget-specific variables and methods, which I've been trying to keep for easy debugging and ensuring merges don't go sideways.
Using :ParticleTracksWidget here stops the tests from running, as it appears that when the test linter goes through it doesn't pick up the TYPE_CHECKING statement and throws a "ParticleTracksWidget not recognised" error.